actions/issue/add-to-project/entrypoint

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -euo pipefail
+shopt -s inherit_errexit
+
+# Github CLI does not currently allow interacting with the API of new-style projects.
+# Github GraphQL API guide: https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project/using-the-api-to-manage-projects
+
+function main() {
+  declare issue_node_id project_org project_num field_name option_name
+  declare -i project_num
+
+  while [[ "${#}" != 0 ]]; do
+    case "${1}" in
+      --token)
+        token="${2}"
+        shift 2
+        ;;
+
+      --issue-node-id)
+        issue_node_id="${2}"
+        shift 2
+        ;;
+
+      --project-org)
+        project_org="${2}"
+        shift 2
+        ;;
+
+      --project-num)
+        project_num=${2}
+        shift 2
+        ;;
+
+      --field-name)
+        field_name="${2}"
+        shift 2
+        ;;
+
+      --option-name)
+        option_name="${2}"
+        shift 2
+        ;;
+
+      "")
+        # skip if the argument is empty
+        shift 1
+        ;;
+
+      *)
+        echo "Unknown argument" "$@"
+        exit 1
+    esac
+  done
+
+  echo "${token}" | gh auth login --with-token
```

Comment:
FYI we have two established patterns for providing authentication to `gh auth`. This is one pattern, and the other is to set the environment variable `$GITHUB_TOKEN` in the action config.

There are pros and cons to both approaches, and I don't think you should change this script to the model. I think I'd rather have a holistic discussion about which pattern we prefer and change all the scripts at once to which pattern we opt for.

I'm just making you aware in case you didn't already know about the other option.